ownCloud/Client/Actions/Scanner/ScanViewController.swift

```diff
@@ -0,0 +1,526 @@
+		}
+	}
+
+	@objc func cancel() {
+		self.dismiss(animated: true, completion: nil)
+	}
+
+	@objc func save() {
+		if let fileName = fileNameRow?.value as? String, let pages = pagesCollectionViewController?.pages, let exporter = self.exportFormat?.exporter, let core = core {
+			let progressHUDViewController = ProgressHUDViewController(on: self, label: "Saving".localized)
+
+			DispatchQueue.global(qos: .userInitiated).async {
+				let tmpURL = URL(fileURLWithPath: NSTemporaryDirectory()).appendingPathComponent(UUID().uuidString)
+
+				try? FileManager.default.createDirectory(at: tmpURL, withIntermediateDirectories: true, attributes: nil)
+
+				var exportedURLs : [URL]?
+
+				if self.exportFormat?.supportsMultiPage == true, (self.oneFilePerPageRow?.value as? Bool) == true {
+					for page in pages {
+						if let exportedPageURLs = exporter(tmpURL, [page]) {
+							if exportedURLs == nil {
+								exportedURLs = []
+							}
+							exportedURLs?.append(contentsOf: exportedPageURLs)
+						} else {
+							exportedURLs = nil
+							break
+						}
+					}
+				} else {
+					exportedURLs = exporter(tmpURL, pages)
+				}
+
+				Log.log("ExportPDF=\(String(describing: exportedURLs))")
+
+				if let targetItem = self.targetFolderItem, let exportedURLs = exportedURLs {
+					for exportedURL in exportedURLs {
+						core.importFileNamed(fileName, at: targetItem, from: exportedURL, isSecurityScoped: false, options: [ .automaticConflictResolutionNameStyle : OCCoreDuplicateNameStyle.bracketed.rawValue ], placeholderCompletionHandler: nil, resultHandler: nil)
+					}
+				}
+
+				OnMainThread {
+					progressHUDViewController.dismiss(animated: true, completion: {
+						self.dismiss(animated: true)
+						try? FileManager.default.removeItem(at: tmpURL)
+					})
+				}
+			}
+		}
+	}
+
+}
+
+@available(iOS 13.0, *)
+extension VNDocumentCameraScan {
+	var scannedPages : [ScanPage] {
+		var pages : [ScanPage] = []
+
+		for pageIdx in 0 ..< self.pageCount {
+			let image = self.imageOfPage(at: pageIdx)
+
+			pages.append(ScanPage(with: image))
+		}
+
+		return pages
+	}
+}
+
+@available(iOS 13.0, *)
+class Scanner : NSObject, VNDocumentCameraViewControllerDelegate {
+	typealias ScannerCompletionHandler = (_ scanner: Scanner, _ error: Error?, _ scan: VNDocumentCameraScan?) -> Void
+
+	var completionHandler : ScannerCompletionHandler?
+
+	static func scan(on viewController: UIViewController, completion: @escaping ScannerCompletionHandler) {
+		let scanner = Scanner()
+
+		scanner.scan(on: viewController, completion: { (_, error, scan) in
+			completion(scanner, error, scan)
+		})
+	}
+
+	func scan(on viewController: UIViewController, completion: @escaping ScannerCompletionHandler) {
+		completionHandler = completion
+
+		let documentCameraViewController = VNDocumentCameraViewController()
+		documentCameraViewController.delegate = self
+		viewController.present(documentCameraViewController, animated: true)
+	}
+
+	@available(iOS 13.0, *)
```

Comment:
@felix-schwarz do you need these `available` constructs since the class is already prepended with `available (iOS 13)`?

Reply:
Good catch! Removed the unneeded ones.